src/v/cluster/rm_stm.cc

```diff
@@ -2618,6 +2621,40 @@ ss::future<> rm_stm::handle_eviction() {
       });
 }
 
+void rm_stm::spawn_background_clean_for_pids(rm_stm::clear_type type) {
+    switch (type) {
+    case rm_stm::clear_type::tx_pids: {
+        if (_log_state.fence_pid_epoch.size() <= _max_saved_pids_count()) {
+            return;
+        }
+        break;
+    }
+    case rm_stm::clear_type::idempotent_pids: {
+        if (_inflight_requests.size() <= _max_saved_pids_count()) {
```

Comment:
`_inflight_requests` is about the inflight requests, it's almost empty while `_log_state.seq_table` contains all pids ever used, we need to clean it